core/src/main/java/com/capitalone/dashboard/client/RestClient.java

```diff
@@ -0,0 +1,198 @@
+        return restOperations.exchange(url, HttpMethod.POST, new HttpEntity<Object>(body, null), String.class);
+    }
+
+    public ResponseEntity<String> makeRestCallPost(String url, String headerKey, String token, JSONObject body) {
+        if (restOperations == null) { return null; }
+
+        if (StringUtils.isEmpty(headerKey) || StringUtils.isEmpty(token)) {
+            return makeRestCallPost(url, body);
+        }
+        return restOperations.exchange(url, HttpMethod.POST, new HttpEntity<Object>(body, createHeaders(headerKey, token)), String.class);
+    }
+
+    public ResponseEntity<String> makeRestCallPost(String url, RestUserInfo userInfo, JSONObject body) {
+        if (restOperations == null) { return null; }
+
+        if ((userInfo == null)) {
+            return makeRestCallPost(url, body);
+        }
+        return restOperations.exchange(url, HttpMethod.POST, new HttpEntity<Object>(body, createHeaders(userInfo.getFormattedString())), String.class);
+
+    }
+
+    public ResponseEntity<String> makeRestCallGet(String url) throws RestClientException {
+        if (restOperations == null) { return null; }
+
+        return restOperations.exchange(url, HttpMethod.GET, null, String.class);
+    }
+
+    public ResponseEntity<String> makeRestCallGet(String url, String headerKey, String token) throws RestClientException {
+        if (restOperations == null) { return null; }
+
+        if (StringUtils.isEmpty(headerKey) || StringUtils.isEmpty(token)) {
+            return makeRestCallGet(url);
+        }
+        return restOperations.exchange(url, HttpMethod.GET, new HttpEntity<>(createHeaders(headerKey, token)), String.class);
+    }
+
+    public ResponseEntity<String> makeRestCallGet(String url, RestUserInfo userInfo) throws RestClientException {
+        if (restOperations == null) { return null; }
+
+        if ((userInfo == null) || StringUtils.isEmpty(userInfo.getFormattedString())) {
+            return makeRestCallGet(url);
+        }
+        return restOperations.exchange(url, HttpMethod.GET, new HttpEntity<>(createHeaders(userInfo.getFormattedString())), String.class);
+    }
+
+    protected HttpHeaders createHeaders(RestUserInfo restUserInfo) {
+        return createHeaders(restUserInfo.getFormattedString());
+    }
+
+    protected HttpHeaders createHeaders(String user) {
+        byte[] encodedAuth = Base64.encodeBase64(user.getBytes(StandardCharsets.US_ASCII));
+        String authHeader = "Basic " + new String(encodedAuth);
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", authHeader);
+        return headers;
+    }
+
+    protected HttpHeaders createHeaders(String key, String token) {
+        String authHeader = key.trim() + " " + token.trim();
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", authHeader);
+        return headers;
+    }
+
+    public JSONObject parseAsObject(ResponseEntity<String> response) throws ParseException {
```

Comment:
all the standalone utility methods that do not depend on restoperations should be static methods and if possible move it to a separate utility class.

Reply:
While in theory it sounds good, but I am afraid practically it presents challenges when building our unit tests around static methods. Mockito doesn't work very well with static methods, thanks